doc/policy/version3_transactions.md

```diff
@@ -0,0 +1,16 @@
+# Transactions with nVersion 3
+
+A transaction with its `nVersion` field set to 3 ("V3 transactions") are allowed in mempool and
+transaction relay.
+
+The following set of policies apply to V3 transactions:
+
+1. Any descendant of an unconfirmed V3 transaction must also be V3.
+
+2. A V3 transaction can be replaced by another V3 transaction, even if it does not signal BIP125
+   replaceability. Use the (`-mempoolfullrbf`) configuration option to allow transaction
+   replacement without enforcement of any opt-in signaling rule.
+
+3. A V3 transaction cannot have more than 1 descendant.
+
+4. A V3 transaction that has a V3 ancestor cannot be larger than 4000 virtual bytes.
```

Comment:
Maybe this is a naive question, but is there a way we can make that value configurable instead of using a protocol hard-coded value?

Informally, any v3 transaction would include a _descendant_max_weight_ somewhere (handwave, handwave), covered by signatures. This lets L2 contracts decide what values are acceptable for each tx type. For example, in the anchor outputs LN-penalty case, we know that commit txs only need an anchor child that can be constrained to be quite small, so commit txs could use a low value here, that can take into account the maximum weight of a commit tx.

Or would this be adding too much complexity for only a marginal improvement? If so, we should probably bikeshed the protocol hard-coded value and use the lowest value we can to minimize the harm a malicious child tx can do.

Reply:
It's definitely doable under an annex field regime(input commits to total package size), but it's a question of complexity.

I do think this kind of system would allow ANYONECANPAY style contracts to be secure against pinning, so it's definitely worth further exploration.